Docker Desktop 4.21.0 release notes (#17622)
docker/docs: desktop/release-notes.md

## 4.21.0

{% include release-date.html date="2023-06-29" %}

{% include desktop-install.md all=true version="4.21.0" build_path="/" %}

### New

- Added support for new Wasm runtimes: slight, spin, and wasmtime.  Users can download Wasm runtimes on demand when the containerd image store is enabled.
- Add Rust server support to Docker init.

### Upgrades
- [Buildx v0.11.0](https://github.com/docker/buildx/releases/tag/v0.11.0)
- [Compose v2.19.0](https://github.com/docker/compose/releases/tag/v2.19.0)
- [Kubernetes v1.27.2](https://github.com/kubernetes/kubernetes/releases/tag/v1.27.2)
- [cri-tools v1.27.0](https://github.com/kubernetes-sigs/cri-tools/releases/tag/v1.27.0)
- [cri-dockerd v0.3.2](https://github.com/Mirantis/cri-dockerd/releases/tag/v0.3.2)
- [coredns v1.10.1](https://github.com/coredns/coredns/releases/tag/v1.10.1)
- [cni v1.2.0](https://github.com/containernetworking/plugins/releases/tag/v1.2.0)
- [etcd v3.5.7](https://github.com/etcd-io/etcd/releases/tag/v3.5.7)

### Bug fixes and enhancements

#### For all platforms

- Automatically pause Docker when idle and wake it up again on demand.

> - VirtioFS will be enabled by defailt in new installations of Docker Desktop on macOS 12.5 and higher.

Reviewer: ```suggestion
- VirtioFS will be enabled by default in new installations of Docker Desktop on macOS 12.5 and higher.
```

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.21.0

{% include release-date.html date="2023-06-29" %}

{% include desktop-install.md all=true version="4.21.0" build_path="/" %}

> ### New

Reviewer 2: I think we should add a note about the CLI hints being added in this release (related to https://github.com/docker/compose-cli/pull/2252). And also tell users how to disable them using the env var `DOCKER_CLI_HINTS=false`. cc @eunomie

Author: added below! Thanks

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.21.0

{% include release-date.html date="2023-06-29" %}

{% include desktop-install.md all=true version="4.21.0" build_path="/" %}

### New

- Added support for new Wasm runtimes: slight, spin, and wasmtime.  Users can download Wasm runtimes on demand when the containerd image store is enabled.
- Added Rust server support to Docker init.
- New [Builds view](use-desktop/builds.md) that lets you inspect builds and manage builders.

> 

Author: Updated in PR!

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.21.0

{% include release-date.html date="2023-06-29" %}

{% include desktop-install.md all=true version="4.21.0" build_path="/" %}

### New

- Added support for new Wasm runtimes: slight, spin, and wasmtime.  Users can download Wasm runtimes on demand when the containerd image store is enabled.
- Added Rust server support to Docker init.

> - New [Builds view](use-desktop/builds.md) that lets you inspect builds and manage builders.

Reviewer 2: @dvdksn I forgot about this but maybe we should say that it's an experimental feature and can be enabled in "Features in development" in Docker Desktop settings?

Reviewer 3: Yeah we could just add "(beta)" after the link I think.

Author: How about? 
```suggestion
- New [Builds view](use-desktop/builds.md) (Beta) that lets you inspect builds and manage builders. This can be found in the **Features in Development** tab in Docker Desktop.
```

Author: Updated the PR!

> - New [Builds view](use-desktop/builds.md) that lets you inspect builds and manage builders.

Reviewer 4: You forgot to remove this 🙈

> - New [Builds view](use-desktop/builds.md) (Beta) that lets you inspect builds and manage builders. This can be found in the **Features in Development** tab in Docker Desktop.

Reviewer 4: ```suggestion
- Beta release of the **Builds** view that lets you inspect builds and manage builders. This can be found in the **Features in Development** tab in **Settings**.
```

Follows a similar pattern we've followed before. I've also removed the link to the page (that isn't published yet) so we can get the PR merged. It's easier for me to retrospectively add links after merge

Author: Right! That makes sense! I've updated it accordingly :)

- You can now reclaim disk space more quickly when files are deleted in containers. Related to [docker/for-mac#371](https://github.com/docker/for-mac/issues/371).
- Fixed a bug that prevented containers accessing 169.254.0.0/16 IPs. Fixes [docker/for-mac#6825](https://github.com/docker/for-mac/issues/6825).
- Fixed a bug in `com.docker.diagnose check` where it would complain about a missing vpnkit even when vpnkit is not expected to be running. Related to [docker/for-mac#6825](https://github.com/docker/for-mac/issues/6825).

> - Fixed a bug where extension menu wasn't clickable. Fixes [docker/for-mac#6840](https://github.com/docker/for-mac/issues/6840) and [docker/for-mac#6855](https://github.com/docker/for-mac/issues/6855)

Reviewer: Are these really forgotten fixes and features we missed in `4.20.1` release notes? 😬

Author: No, I misplaced the release note!
